src/main/scala/MarkovChain.scala

```diff
@@ -0,0 +1,67 @@
+package uk.gov.hmrc.smartstub
+
+import org.scalacheck.Gen
+import org.scalacheck.Gen._
+import cats.implicits._
+import org.scalacheck.support.cats._
+import java.util.NoSuchElementException
+
+/* 
+ * A MarkovChain implementation that will return a plausible/probable
+ * next item given a seed sub-sequence from the original data. Ideally
+ * the seed should be windowSize length but it will return something
+ * based on any seed (including empty) as long as the non-empty seed
+ * exists in the original data. 
+ * 
+ * If a terminus is supplied it will use this as seed whenever an
+ * empty seed is supplied.
+ */
+class MarkovChain[A](
+  val data: Seq[A],
+  val windowSize: Int,
+  val terminus: Seq[A] = Nil
+) {
+
+  val multimap: Map[Int, Map[Seq[A], Gen[A]]] = {
+    (for (i <- 1 to windowSize) yield i -> markov(i)).toMap
+  }
+
+  private def start:Seq[A] = {
```

Comment:
this can be a `val` - then at least we would see if `.head` throws as soon as this class is instantiated